tests/end2end/features/misc.feature

```diff
@@ -671,3 +671,38 @@ Feature: Various utility commands.
         And I wait until the scroll position changed
         And I run :click-element id link
         Then the error "Element position is out of view!" should be shown
+
+    ## :command-history-{prev,next}
+    
+    Scenario: Calling previous command
+        When I run :set-cmd-text :message-info blah
+        And I run :command-accept
+        And I wait for "blah" in the log
+        And I run :set-cmd-text :
+        And I run :command-history-prev
+        And I run :command-accept
+        Then the message "blah" should be shown
+    
+    Scenario: Browsing through commands 
+        When I run :set-cmd-text :message-info blarg
+        And I run :command-accept
+        And I wait for "blarg" in the log
+        And I run :set-cmd-text :
+        And I run :command-history-prev
+        And I run :command-history-prev
```

Comment:
I can reproduce that issue locally, but I don't get what's going on - chances are you found a bug somwhere, or the test has some race condition I didn't spot yet :wink:

The repeated command prints the "Hello World" from earlier, and I could reproduce this with this minimized `misc.feature`:

```gherkin
Feature: Various utility commands.

    Scenario: Partial commandline matching
        When I run :set-cmd-text :message-i "Hello World"
        And I run :command-accept
        Then the message "Hello World" should be shown

    Scenario: Sharing cookies with private browsing
        When I set general -> private-browsing to true
        And I open cookies/set?qute-test=42 without waiting
        And I wait until cookies is loaded
        And I open cookies in a new tab
        Then the cookie qute-test should be set to 42

    Scenario: Calling previous command
        When I run :set-cmd-text :message-info blah
        And I run :command-accept
        And I wait for "blah" in the log
        And I run :set-cmd-text :
        And I run :command-history-prev
        And I run :command-accept
        Then the message "blah" should be shown
    
    Scenario: Browsing through commands 
        When I run :set-cmd-text :message-info blarg
        And I run :command-accept
        And I wait for "blarg" in the log
        And I run :set-cmd-text :
        And I run :command-history-prev
        And I run :command-history-prev
        And I run :command-history-next
        And I run :command-accept
        Then the message "blarg" should be shown
```

I'll try to take a closer look tomorrow, but I'm quite busy currently...